src/darwin/Framework/CHIP/MTROTAUnsolicitedBDXMessageHandler.mm

```diff
@@ -0,0 +1,142 @@
+/**
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import "MTROTAUnsolicitedBDXMessageHandler.h"
+
+#include <protocols/bdx/BdxMessages.h>
+
+using namespace chip;
+using namespace chip::Messaging;
+using namespace chip::bdx;
+
+MTROTAUnsolicitedBDXMessageHandler * MTROTAUnsolicitedBDXMessageHandler::sInstance = nullptr;
+
+CHIP_ERROR MTROTAUnsolicitedBDXMessageHandler::Init(ExchangeManager * exchangeManager)
+{
+    assertChipStackLockedByCurrentThread();
+
+    VerifyOrReturnError(exchangeManager != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    MTROTAUnsolicitedBDXMessageHandler::mNumberOfDelegates = 0;
+    mExchangeMgr = exchangeManager;
+    return mExchangeMgr->RegisterUnsolicitedMessageHandlerForProtocol(Protocols::BDX::Id, this);
+}
+
+MTROTAUnsolicitedBDXMessageHandler * MTROTAUnsolicitedBDXMessageHandler::GetInstance()
+{
+    return sInstance;
+}
+
+void MTROTAUnsolicitedBDXMessageHandler::Shutdown()
+{
+    assertChipStackLockedByCurrentThread();
+
+    VerifyOrReturn(mExchangeMgr != nullptr);
+
+    mExchangeMgr->UnregisterUnsolicitedMessageHandlerForProtocol(Protocols::BDX::Id);
+    MTROTAUnsolicitedBDXMessageHandler::mNumberOfDelegates = 0;
+
+    if (mOTAImageTransferHandler != nullptr) {
+        mOTAImageTransferHandler->DestroySelf();
+        delete mOTAImageTransferHandler;
```

Comment:
This ... this is a double-delete.  With some allocators, this will crash.  Please do not do that.  Just do the delete, take out the DestroySelf call, DestroySelf should not be public API as mentioned elsewhere.

What _does_ need to happen and what this code is not doing: `mOTAImageTransferHandler = nullptr;`

Reply:
removed call to DestroySelf and setting mOTAImageTransferHandler = nullptr